src/Microsoft.DotNet.Cli.Utils/PathUtility.cs

```diff
@@ -198,5 +199,43 @@ public static string GetPathWithDirectorySeparator(string path)
                 return GetPathWithBackSlashes(path);
             }
         }
+
+        public static string GetProjectRootPath(string path = null)
+        {
+            var directoryPath = Directory.GetCurrentDirectory();
+
+            if (path != null)
+            {
+                try
+                {
+                    directoryPath = Path.GetDirectoryName(directoryPath);
+                }
+                catch (ArgumentException)
+                {
+                    throw new ArgumentException("Illegal Characters in Path", nameof(directoryPath));
+                }
+            
+                if(!Directory.Exists(directoryPath))
+                {
+                    throw new ArgumentException($"'{directoryPath}' does not exist.", nameof(directoryPath));
+                }
+            }
+
+            string projectRootPath = null;
+
+            directoryPath = directoryPath ?? Directory.GetCurrentDirectory();
```

Comment:
isn't this already set to current directory? can Path.GetDirectoryName return null?